ic10emu/src/grammar.rs

```diff
@@ -765,9 +797,17 @@ impl FromStr for Operand {
                         val.get_str("value").unwrap().parse().unwrap(),
                     )))
                 } else if let Ok(lt) = LogicType::from_str(s) {
-                    Ok(Operand::LogicType(lt))
+                    if let Ok(slt) = SlotLogicType::from_str(s) {
+                        Ok(Operand::LogicOrSlotLogicType(lt, slt))
+                    } else {
+                        Ok(Operand::LogicType(lt))
+                    }
                 } else if let Ok(slt) = SlotLogicType::from_str(s) {
-                    Ok(Operand::SlotLogicType(slt))
+                    if let Ok(lt) = LogicType::from_str(s) {
+                        Ok(Operand::LogicOrSlotLogicType(lt, slt))
```

Comment:
I think this branch is unreachable, since we already checked for `if let Ok(lt) = LogicType::from_str(s)` above

Reply:
It might actually be, but my brain doesn't like not keeping the check...